apps/allocations/app/components/App/PayoutsTables.js

```diff
@@ -4,21 +4,25 @@ import styled from 'styled-components'
 
 import { Badge, Text, theme } from '@aragon/ui'
 
-const PayoutsTable = ({ list: List, data, ...listProps }) =>
-  <div>
-    <Text.Block size="large" weight="bold" style={{ marginBottom: '10px' }}>
-      Allocations
-      {' '}
-      <Badge.Info>{data.length}</Badge.Info>
-    </Text.Block>
+const PayoutsTable = props => {
+  const List = props.list
 
-    <List {...listProps} />
-  </div>
+  return (
+    <div>
+      <Text.Block size="large" weight="bold" style={{ marginBottom: '10px' }}>
+        Allocations
+        {' '}
+        <Badge.Info>{props.data.length}</Badge.Info>
+      </Text.Block>
+
+      <List {...props} />
```

Comment:
This now passes _all_ props to `List`, including `list` and `data`. We didn't pass it those two props, previously.

Why change the argument/prop signature on this component? Which linting rule complained about the old one?

Reply:
Linting didn't complain here, but props were not correctly passed and PayoutTable wasn't working - listProps was undefined. I didn't investigate too much, and the most concise way of making it work again was to tweak what was already there. I think only "list" is superfluous - I can exclude it, so code is more explicit..?

Reply:
Looks like we are calling this with:

```jsx
<PayoutsTable
  data={sortedPayouts}
  tokens={tokens}
  executePayout={executePayout}
  list={PayoutsNarrow}
/>
```

Originally, the code you've changed here had this:

```jsx
const PayoutsTable = ({ list: List, data, ...listProps }) =>
```

Which means `listProps` would have been an object like:

```js
{ tokens, executePayout }
```

Going back to `PayoutsNarrow`, it looks like it expects the following args:

```js
const PayoutsNarrow = ({ executePayout, data, tokens }) => (
```

Since the previous code was only expanding `listProps`, this component would only have gotten passed `executePayout` and `tokens`. It probably broke because it wasn't being passed `data`.

Rather than changing this whole function, I think we only needed to start passing `data` to `List`:

```jsx
<List data={data} {...listProps} />
```

Reply:
Ok, I see the point. There is one more posibility: `data`, `tokens` and `executePayout` are needed all the way through, but `PayoutsTable` needs data only to check its length. So how about:
```
const PayoutsTable = ({ list: List, ...listProps }) => (
  <div>
(...)
      <Badge.Info>{listProps.data.length}</Badge.Info>
(...)
    <List {...listProps} />
  </div>
)
```
This is even cleaner, at least to me - would you agree?

Reply:
Yes, I like that option, too.